hazelcast/src/main/java/com/hazelcast/topic/impl/TotalOrderedTopicProxy.java

```diff
@@ -47,4 +62,53 @@ public void publish(@Nonnull  E message) {
         InternalCompletableFuture f = invokeOnPartition(operation);
         f.joinInternal();
     }
+
+    @Override
+    public InternalCompletableFuture<Void> publishAsync(@Nonnull E message) {
+        checkNotNull(message, NULL_MESSAGE_IS_NOT_ALLOWED);
+        Operation op = new PublishOperation(getName(), toData(message)).setPartitionId(partitionId);
```

Comment:
No need to set the partition ID if it'll be invoked on the partition by the helper method.

Reply:
removed